### PR TITLE
Disallow updates of Queue arguments

### LINKED ISSUE #822: Queue arguments are not re-concealed
## Describe the bug
Queue arguments are not being re-concealed after initial values are set.

## To Reproduce
1. define x-message-ttl to let's say 30000
2. apply
3. change x-message-ttl to 60000
4. change is not being applied

```yaml
apiVersion: rabbitmq.com/v1beta1
kind: Queue
metadata:
  name: jobs-delayed
  namespace: test
spec:
  name: "jobs_delayed"
  vhost: test
  type: quorum
  arguments:
    x-message-ttl: 60000
    x-dead-letter-exchange: "xchg_jobs"
    x-dead-letter-routing-key: "job"
  autoDelete: false
  durable: true
  rabbitmqClusterReference:
    name: rabbitmq
```

**Expected behavior**
x-message-ttl set to 60000

## Version and environment information
- Messaging Topology Operator: bitnami/rmq-messaging-topology-operator:1.14.1-debian-12-r1
- RabbitMQ: bitnami/rabbitmq:3.13.2-debian-12-r4
- RabbitMQ Cluster Operator: bitnami/rabbitmq-cluster-operator:2.9.0-debian-12-r1
- Kubernetes: v1.29.4+k3s1

### PR DESCRIPTION
**Pull request by Zerpet**
This closes #822

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed
**Note to contributors:** remember to re-generate client set if there are any API changes

## Summary Of Changes

Queue arguments can't be updated in RabbitMQ. The webhook now disallows updates on queue arguments, matching the behaviour of RabbitMQ.

### CODE REVIEW

File: api/v1beta1/queue_webhook_test.go [hunk 1]
```diff
@@ -130,5 +132,27 @@ var _ = Describe("queue webhook", func() {
...
+		It("does not allow updates on queue arguments", func() {
+			By("not allowing value changes, nor adding new values")
+			newQueue := queue.DeepCopy()
+			newQueue.Spec.Arguments = &runtime.RawExtension{
+				Raw: []byte(`{"this-argument": "really cant be updated", "adding-args": "not possible"}`),
+			}
+			_, err := newQueue.ValidateUpdate(rootCtx, &queue, newQueue)
+			Expect(err).To(MatchError(ContainSubstring("queue arguments cannot be updated")))
+
+			By("now allowing removal")
```
Comment (DanielePalaia) [inline api/v1beta1/queue_webhook_test.go:145]: not allowing

File: Makefile [hunk 2]
```diff
@@ -51,23 +51,27 @@ $(KUBEBUILDER_ASSETS):
 
 .PHONY: unit-tests
 unit-tests::install-tools ## Run unit tests
-unit-test::$(KUBEBUILDER_ASSETS)
-unit-test::generate
-unit-test::fmt
-unit-test::vet
-unit-test::vuln
```
Comment (DanielePalaia) [inline Makefile:58]: is vuln removed now?

**Review by DanielePalaia — APPROVED** [review]

File: api/v1beta1/queue_webhook_test.go (hunk 1 above)
Comment (Zerpet) [reply api/v1beta1/queue_webhook_test.go]: Fixed, good catch

File: Makefile (hunk 2 above)
Comment (Zerpet) [reply Makefile:58]: I split in two commits the code changes and the Makefile: f13f3597960e9240e99c1d71ec300385f905ba0e

In summary, I don't think there's much value in running `vuln` during unit tests. The `vuln` scanner still is available to run manually at any time.

**Pull request merged**